runtime/lib/DataAPI/localDataAPI.ts

```diff
@@ -28,7 +28,7 @@ class LocalDataAPI<
 
   public getVersion = async () => this.version;
 
-  public getProgram = async (programID: string) => this.programs[programID];
+  public getProgram = async (_versionID: string, diagramID: string) => this.programs[diagramID];
```

Comment:
I assume `diagramID` is not unique (it's `diagramID` field not `_id`) here, right? if so we should cache programs per version just to be extra safe
```suggestion
  public getProgram = async (versionID: string, diagramID: string) => this.programs[versionID]?.[diagramID] ?? null;
```